### PR TITLE
try to bring in controller tests: introduce index action tests, show action tests

### PR DESCRIPTION
**Pull request by tildeee**
## Description
All the details are here: everything in this PR is probably checked off on this card.
- recontexutalize routing
- The Rails Res/Req Cycle lecture mentions the show action before we do the show action. Smooth that over
- Create resource: Adding Tests to Index
- Create resource: Intro to the Show action and its tests
- Smooth over: Intro to the Show action and its tests -> Livecode

## Questions for the Author
Need to do all of this
- [ ] Have you updated the weekly learning goals in the pedagogy resource?
- [ ] Have you updated the weekly assessment in Schoology?
- [ ] Have you updated the READMEs with links to new or moved resources?

## Todos
- Figure out how to teach testing the show action without a test db

## Feedback Requested
Just realized that testing show, update, and delete would be really hard if they don't have knowledge about the test db/fixtures yet...

### CODE REVIEW

File: 08-rails/controllers.md [hunk 1]
```diff
@@ -13,7 +13,13 @@ A Rails Controller is the central manager of a Rails application. It takes reque
...
 
-The things that a controller can do are called _actions_, and each action will be an instance method on a controller class. For example, our `BooksController` might have an `index` action that sends back a list of books to the browser, and a `show` action that sends back details on one book. Each of those would be an instance method in the `BooksController`.
+The things that a controller can do are called _actions_, and each action will be an instance method on a controller class.
+
+We can, and eventually will, define our own custom _actions_, so that we can define more and more things for our controller to do. However, for today, we will be covering a lot of controller actions that are _Rails conventions_.
+
+The first two controller actions that we will introduce are the _index_ action and the _show_ action. Our _index_ action will usually mean that our controller will send back some important information. Our _show_ action will usually mean that our controller will send back details of one thing.
```
Comment (kaidamasaki) [inline 08-rails/controllers.md]: Is it possible to be more specific than "important information" for _index_?

Comment (kaidamasaki) [reply 08-rails/controllers.md]: Format "index" and "show" as code like below?

```suggestion
The first two controller actions that we will introduce are the `index` action and the `show` action. Our `index` action will usually mean that our controller will send back some important information. Our `show` action will usually mean that our controller will send back details of one thing.
```

File: 08-rails/intro-to-testing-controllers.md [hunk 2]
```diff
@@ -65,13 +64,24 @@ end
 ```
 Notice that we are **not** testing the body content of the response.  The particular HTML page in the response is likely to change and is difficult and expensive to test.  Instead controller tests focus on the bigger picture including response code, and changes to the database.
 
-### Testing The Show, Edit & New Actions
+## Nominal Cases & Edge Cases for `index`
```
Comment (kaidamasaki) [inline 08-rails/intro-to-testing-controllers.md]: I like the reinforcing "nominal" and "edge" cases here. 😃
